// @qdrant/js-client-grpc/dist/cjs/proto/snapshots_service_connect.js
"use strict";
// @generated by protoc-gen-connect-es v0.10.0 with parameter "target=ts"
// @generated from file snapshots_service.proto (package qdrant, syntax proto3)
/* eslint-disable */
// @ts-nocheck
Object.defineProperty(exports, "__esModule", { value: true });
exports.Snapshots = void 0;
const snapshots_service_pb_js_1 = require("./snapshots_service_pb.js");
const protobuf_1 = require("@bufbuild/protobuf");
/**
 * @generated from service qdrant.Snapshots
 */
exports.Snapshots = {
    typeName: "qdrant.Snapshots",
    methods: {
        /**
         *
         * Create collection snapshot
         *
         * @generated from rpc qdrant.Snapshots.Create
         */
        create: {
            name: "Create",
            I: snapshots_service_pb_js_1.CreateSnapshotRequest,
            O: snapshots_service_pb_js_1.CreateSnapshotResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         *
         * List collection snapshots
         *
         * @generated from rpc qdrant.Snapshots.List
         */
        list: {
            name: "List",
            I: snapshots_service_pb_js_1.ListSnapshotsRequest,
            O: snapshots_service_pb_js_1.ListSnapshotsResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         *
         * Delete collection snapshot
         *
         * @generated from rpc qdrant.Snapshots.Delete
         */
        delete: {
            name: "Delete",
            I: snapshots_service_pb_js_1.DeleteSnapshotRequest,
            O: snapshots_service_pb_js_1.DeleteSnapshotResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         *
         * Create full storage snapshot
         *
         * @generated from rpc qdrant.Snapshots.CreateFull
         */
        createFull: {
            name: "CreateFull",
            I: snapshots_service_pb_js_1.CreateFullSnapshotRequest,
            O: snapshots_service_pb_js_1.CreateSnapshotResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         *
         * List full storage snapshots
         *
         * @generated from rpc qdrant.Snapshots.ListFull
         */
        listFull: {
            name: "ListFull",
            I: snapshots_service_pb_js_1.ListFullSnapshotsRequest,
            O: snapshots_service_pb_js_1.ListSnapshotsResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
        /**
         *
         * Delete full storage snapshot
         *
         * @generated from rpc qdrant.Snapshots.DeleteFull
         */
        deleteFull: {
            name: "DeleteFull",
            I: snapshots_service_pb_js_1.DeleteFullSnapshotRequest,
            O: snapshots_service_pb_js_1.DeleteSnapshotResponse,
            kind: protobuf_1.MethodKind.Unary,
        },
    }
};
